### DISCUSSION TITLE
reverting Github commits

### DISCUSSION THREAD

**Discussion by Aiden-Demacio** [community/community#70190] [New to GitHub]
### Select Topic Area

Question

### Body

In a group project I am involved in, the first time a member of my group commited something, the whole thing broke. Another group member and I are working to resolve this issue and trying to revert the changes but are getting lost as to how to do so as files have been overwritten. How would we go about reverting back to when this worked?

**Comment by seyLu** [comment]
#### Hi @Aiden-Demacio,

First, list out the commits:

```bash
git log --oneline
```

Copy the commit SHA (the 7-character combination of letters and numbers before the commit message) of the commit you wish to undo.

Then revert that commit

```bash
git revert <commit_SHA>
```

This will undo the changes made by that specific commit.

Add, commit, push as usual.


---

<br>

If you want to undo multiple changes since a commit -- going back in history and undoing all the commits made up to that point.

Then git reset would make more sense. Again, list out the commits.

```bash
git log --oneline
```

Copy the commit SHA you'd want to go back to in history.

Reset all changes made up to that point.

```bash
git reset <commit_SHA>
```

Add, commit, push as usual.

<br>
<br>

You can read more at:
- [Git docs: git revert](https://git-scm.com/docs/git-revert)
- [Git docs: git reset](https://git-scm.com/docs/git-reset)
